pyPhin.py

```diff
@@ -109,47 +118,80 @@ def verify(self, contact, verifyUrl, deviceUUID, verificationCode):
 		'''
 		req = requests.get(
 			self.baseUrl+locationUrl,
-			headers=self.createHeader(authToken, deviceUUID, "2.0.1")
+			headers=self.createHeader(deviceUUID, authToken, "2.0.1")
 			)
 		reqJson = json.loads(req.text)
 
-		if not reqJson["success"]:
-			raise Exception(reqJson)
+		self.checkRequest(req.text)
 
 		vesselUrl = reqJson["locations"][0]["resources"]["vessels"]["route"]
 
-		auth = {"authToken":authToken,"vesselUrl":vesselUrl,"UUID":deviceUUID}
-		return json.dumps(auth)
+		#Auth Dictionary Structure needed to access data.
+		authData = {"authToken":authToken,"vesselUrl":vesselUrl}
 
-	def createHeader(self, authToken, deviceUUID, version="1.0.0"):
-		tempHeader = {"x-phin-concise":"true",
-			"x-phin-reporting-app-id":"ios-app",
-			"x-phin-reporting-device-id":deviceUUID}
-		tempHeader["Accept-Version"] = version
-		tempHeader["Authorization"] = "Bearer " + authToken
-		return tempHeader
+		return authData
+
+	def getData(self, authToken, deviceUUID, vesselUrl):
 
-	def getData(self, auth, data):
+		self.checkUrlRoute(vesselUrl)
 
-		dataCluster = json.loads(auth)
+		data = {}
 
-		if data.upper() == "TA" or "CYA" or "TH":
-			return self.getWaterQuality(dataCluster["authToken"], dataCluster["vesselUrl"], dataCluster["UUID"], data.upper())
+		data["waterData"] = self.getWaterData(
+			authToken,
+			deviceUUID,
+			vesselUrl)
 
-	def getWaterQuality(self, authToken, vesselUrl, deviceUUID, waterQualityType):
+		return data
+
+	def getWaterData(self, authToken, deviceUUID, vesselUrl):
 
 
 		req = requests.get(
 			self.baseUrl+vesselUrl,
-			headers=self.createHeader(authToken,deviceUUID, "2.0.0")
+			headers=self.createHeader(deviceUUID, authToken, "2.0.0")
 			)
-		reqJson = json.loads(req.text)
-		print(reqJson)
-
 
-		return reqJson["vessels"][0]["waterReport"][waterQualityType]["value"]
 
+		self.checkRequest(req.text)
+		reqJson = json.loads(req.text)
 
+		data = {}
 
+		for dataType in ["TA","CYA","TH"]:
+			data[dataType] = reqJson["vessels"][0]["waterReport"][dataType]["value"]
```

Comment:
What if pHin decide not return value, eg no "CYA" comes back... in that case

data[dataType] = reqJson["vessels"][0]["waterReport"][dataType] == None

and thus your code errors out..

Reply:
Well thats a bit extreme. It would say `KeyError: CYA` , and instead of me typing an exception saying `Exception: Cannot Find CYA`.